### PR TITLE
MGDOBR-1328: Add existing components stories to storybook

### PR DESCRIPTION
**Pull request by kelvah**
JIRA: https://issues.redhat.com/browse/MGDOBR-1328

Added stories for the following components:
- `CreateInstance`
- `DeleteModal`
- `EmptyState`
- `PageNotFound`

Added missing stories to `SEStatusLabel`.

### CODE REVIEW

File: src/app/Instance/CreateInstance/CreateInstance.tsx [hunk 1]
```diff
@@ -31,6 +31,8 @@ export interface CreateInstanceProps {
     onSuccess: () => void,
     onError: (error: CreateInstanceError) => void
   ) => void;
+  /** The parent container to append the dialog to. Defaults to "document.body". */
+  appendTo?: () => HTMLElement;
```
Comment (kelvah) [inline src/app/Instance/CreateInstance/CreateInstance.tsx:35]: I had to add the `appedTo` prop to make sure the dialog is rendered inside storybook preview `div` and then use [play functions](https://storybook.js.org/docs/react/writing-stories/play-function) to simulate user interaction.

File: src/app/components/EmptyState/EmptyState.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+EmptyStateBase.args = {
+  title: "No Smart Event instances",
+  createButton: {
+    title: "Create Smart Events instance",
+    onCreate: (): void => {},
+  },
+  quickStartGuide: {
+    i18nKey: "common.quickStartAccess",
+    onQuickstartGuide: (): void => {},
+  },
```
Comment (kelvah) [inline src/app/components/EmptyState/EmptyState.stories.tsx:25]: For some reason the `i18nKey` is not converted properly to its corresponding string within the `EmptyState` story. I couldn't find why.
I guess It's not a big deal for now. Suggestions are obviously welcome.

File: package.json [hunk 3]
```diff
@@ -39,6 +39,7 @@
     "@rhoas/app-services-ui-shared": "0.16.6",
     "@rhoas/smart-events-management-sdk": "0.51.0",
     "@xstate/react": "3.0.1",
+    "date-fns": "2.29.3",
```
Comment (vpellegrino) [inline package.json:42]: General question about stories.
Are all dependencies used in them considered dev or prod dependencies?

File: src/app/Instance/CreateInstance/storiesHelpers.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+export const sampleSubmit: PlayFunction<
+  ReactFramework,
+  CreateInstanceProps
+> = async ({ canvasElement }) => {
+  const canvas = within(canvasElement);
+
+  await waitFor(
+    () => {
+      // expect(canvas.getByLabelText("Name *")).toBeEnabled();
```
Comment (vpellegrino) [inline src/app/Instance/CreateInstance/storiesHelpers.ts:40]: this commented line can be removed?

**Review by vpellegrino — APPROVED** [review]
The changes look good to me.
Appreciate your effort with stories.
I left just one minor comment.
Thank you, @kelvah

File: src/app/Instance/CreateInstance/storiesHelpers.ts (hunk 4 above)
Comment (kelvah) [reply src/app/Instance/CreateInstance/storiesHelpers.ts]: whoops. done.

File: package.json (hunk 3 above)
Comment (kelvah) [reply package.json:42]: @vpellegrino I would say dev. All storybook deps are `devDependencies`.

`date-fns` is an exception though. It was already [used](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/blob/d95c69d18a3834dea9e4b5aba28ff2c7b10a44fe/src/app/Instance/InstanceDetails/InstanceDetails.tsx#L22) [elsewhere](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/blob/d95c69d18a3834dea9e4b5aba28ff2c7b10a44fe/src/app/components/SEStatusLabel/useResourceDelayedStatus.ts#L6) as transitive dep. 
When importing it inside a story I've finally made things right.

Comment (vpellegrino) [reply package.json:42]: Makes sense. Thank you @kelvah

**Review by ppadti — APPROVED** [review]

**Pull request merged**